### DISCUSSION TITLE
Help needed to make Ventoy work for FreeBSD 13.1 based helloSystem

### DISCUSSION THREAD

**Discussion by probonopd** [ventoy/Ventoy#1721] [Q&A]
Using [Ventoy 1.0.78 release](https://github.com/ventoy/Ventoy/releases/tag/v1.0.78):

* FreeBSD 13.0 based helloSystem works with Ventoy (FreeBSD bootloader says `/boot/kernel/geom_ventoy.ko size 0xc59e0 at 0x213f000`)
* FreeBSD 13.1 based helloSystem [does not work](https://github.com/helloSystem/ISO/issues/385) with Ventoy (FreeBSD bootloader says `don't know how to load module '/boot/kernel/geom_ventoy.ko'` only if the ISO is loaded through Ventoy)
* But FreeBSD 13.1 based GhostBSD works with Ventoy (FreeBSD bootloader does not say anything about `geom_ventoy.ko`)

It seems like the GhostBSD ISO doesn't even contain the Ventoy kernel module, whereas helloSystem does contain it and is using the FreeBSD bootloader to load it (which worked in FreeBSD 13.0 but apparently not in FreeBSD 13.1 when booted through Ventoy).

Can you help @ventoy? Thank you very much.

**Comment by ventoy** [comment]
What about directly use this ko file in your system?
[geom_ventoy.ko.xz](https://github.com/ventoy/Ventoy/blob/master/Unix/ventoy_unix/FreeBSD/geom_ventoy_ko/13.x/64/geom_ventoy.ko.xz)

**Reply by probonopd** [reply]
It is what we are using...

https://github.com/helloSystem/ISO/blob/c0f278e98fe49908ca7024c40ef2de357e7faf1b/build.sh#L500

**Reply by 0hip** [reply]
Good news. Since [hello-0.8.2_0H331-FreeBSD-13.1-amd64.iso](https://github.com/helloSystem/ISO/releases/download/experimental-13.1/hello-0.8.2_0H331-FreeBSD-13.1-amd64.iso) and as long as unmodified FreeBSD ISO succesfully boots through VENTOY (i.e. with no added ventoy module at all, which is actually the case now), hellosystem also succesfully boots.

**Comment by mnaray** [comment]
Make sure that you are booting from an optical drive if the computer is older. Some BIOSes don't like Thumb drives. I had this issue for 5 days, and I was about to have a mental breakdown when someone told me about that.

USB optical drive should work, otherwise try a SATA one. 👍

**Reply by probonopd** [reply]
Interesting! I have never encountered this so far. Which BIOS/manufacturer is affected?

**Reply by mnaray** [reply]
Some older Phoenix Award BIOSes are affected. I don't know the specific versions by heart, but the device I had to do it on was around 13-15 years old.

**Reply by nrm** [reply]
FreeBSD-13.1_memstick recorded on USB(dd) works.
Manufacturer: Supermicro
BIOS: American Megatrends Inc.
Release: 04/09/2015

Ventoy with Freebsd_13.1.iso doesn't work.
